File: src/leetcode/m151/ReverseWordsInAString.java
/*
 * Copyright (c) 2024 THGEEK.COM. All rights reserved.
 */

package leetcode.m151;

/**
 * LeetCode 151
 *
 * @author Tao Hong
 * @version 1.0
 * @since 2024/09/24 10:28
 */
public class ReverseWordsInAString {
    public String reverseWords(String s) {
        String[] words = s.split("\\s+");
        int left = 0;
        int right = words.length - 1;
        while (left < right) {
            String word = words[left];
            words[left] = words[right];
            words[right] = word;
            left++;
            right--;
        }
        return String.join(" ", words);
    }
}

File: src/leetcode/m151/ReverseWordsInAString.go
package m151

import (
	"strings"
)

// LeetCode 151
func reverseWords(s string) string {
	fields := strings.Fields(s)
	for i, j := 0, len(fields)-1; i < j; i, j = i+1, j-1 {
		fields[i], fields[j] = fields[j], fields[i]
	}

	return strings.Join(fields, " ")
}
